snapcraft/internal/sources.py

```diff
@@ -435,3 +476,52 @@ def _get_source_type_from_uri(source, ignore_errors=False):
         raise ValueError('local source is not a directory')
 
     return source_type
+
+
+def verify_checksum(source_checksum, checkfile):
+    if source_checksum.startswith('http'):
+        response = urllib.request.urlopen(source_checksum)
+        data = response.read()
+        source_checksum = data.decode('utf-8')
+        if (" " in source_checksum):
+            source_checksum = source_checksum.split(" ", 1)[0]
+        else:
+            print('No file name detected in the checksum file, perhaps an '
```

Comment:
@sergiusens do you think it's **always** the case that the checksum file has a filename? If so, I'll correct it.